### PR TITLE
Create files and groups

### PR DESCRIPTION
**Pull request by brentleyjones**
They aren't added to the main group yet.

### CODE REVIEW

File: tools/generator/src/Generator+CreateFilesAndGroups.swift [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ) {
+        var elements: [FilePath: PBXFileElement] = [:]
+
+        var externalGroup: PBXGroup?
+        func createGroup(
+            filePath: FilePath,
+            pathComponent: String
+        ) -> PBXGroup {
+            let group: PBXGroup
+            // TODO: Handle in-workspace "external/" paths
```
Comment (brentleyjones) [inline tools/generator/src/Generator+CreateFilesAndGroups.swift:26]: TODO for a future Brentley

File: tools/generator/src/Generator+CreateFilesAndGroups.swift [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            filePath: FilePath,
+            pathComponent: String,
+            isLeaf: Bool
+        ) -> (PBXFileElement, isNew: Bool) {
+            if let element = elements[filePath] {
+                return (element, false)
+            }
+
+            // TODO: Handle localized files
+            // TODO: Handle CoreData models
```
Comment (brentleyjones) [inline tools/generator/src/Generator+CreateFilesAndGroups.swift:55]: Same. Not needed at this exact moment.

**Comment by brentleyjones** [comment]
Current dependencies on/for this PR:
* main
  * **PR #22** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/22" target="_blank">Graphite</a>  (this PR)
    * **PR #23** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/23" target="_blank">Graphite</a> 
      * **PR #24** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/24" target="_blank">Graphite</a> 
        * **PR #25** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/25" target="_blank">Graphite</a> 
          * **PR #26** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/rules_xcodeproj/26" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**